packages/admin/src/common/toolbar/Toolbar.styles.tsx

```diff
@@ -0,0 +1,62 @@
+import { Theme } from "@material-ui/core/styles";
+import { makeStyles } from "@material-ui/styles";
+
+import { useComponentThemeProps } from "../../mui/useComponentThemeProps";
+
+export interface ToolbarThemeProps {
+    elevation?: number;
+}
+
+export type CometAdminToolbarClassKeys = "root" | "muiToolbar" | "mainContentContainer" | "actionContainer";
+
+export const useStyles = makeStyles<Theme, { headerHeight: number }, CometAdminToolbarClassKeys>(
+    () => ({
+        root: {
+            position: "sticky",
+            display: "flex",
+            flexDirection: "column",
+            justifyContent: "center",
+            top: (props) => props.headerHeight,
+            padding: 0,
+            "& [class*='MuiButton']": {
+                textTransform: "none",
+            },
+        },
+        muiToolbar: {
+            display: "flex",
+            flex: 1,
+            alignItems: "stretch",
+        },
+        mainContentContainer: {
+            display: "flex",
+            flex: 1,
+        },
+        actionContainer: {
+            display: "flex",
+            alignItems: "center",
+            /* Margin on all first level children, without last one */
+            "& > *:not(:last-child)": {
+                marginRight: 20,
+            },
```

Comment:
Should positioning and spacing of the action-items maybe be done in the application, not the library?  

20px might not always be exactly the spacing I want for every instance. I might want a different value, depending on the specific action-items on this page, sometimes maybe I just want a 1px border, or no spacing at all. 🤔

If a spacing is needed it could easily be done in the application with a Mui-Grid:

```tsx
<Toolbar 
    actionItems={(stackSwitchApi: IStackSwitchApi | undefined) => {
        return (
            <Grid container spacing={4}>
                <Grid item>
                    <ActionItem />
                </Grid>
                <Grid item>
                    <ActionItem />
                </Grid>
                <Grid item>
                    <ActionItem />
                </Grid>
            </Grid>
        );
    }}
>
    <ToolbarTitleItem />
</Toolbar>
```

Reply:
What about an API similar to this?

```
<Toolbar>
     <ActionsArea spacing={4}>
           <Button/>
              ....
     </ActionsArea>
<Toolbar>         

```

This would avoid the burden of defining the grid in the app, but still brings flexibilty of the adjusting layout.

Reply:
Shouldn't the spacing apply to the whole toolbar?

Reply:
> Should positioning and spacing of the action-items maybe be done in the application, not the library?

No, it should not be that much code necessary.

This new Toolbar will be  used across all comet-admin applications. It should always look and feel the same. It might be slightly adjusted using the theme, but then that should be consistent for the application.

If you need a special, custom toolbar, use plain mui.

Reply:
good questions. i like all your approaches. The styling here should be removed anyways, i already removed it. @jamesricky i just updated the ToolbarBreadcrumb Story with the Grid solution. ❤️ 

Some further thoughts:

i intended to make a toolbar relatively strict to our ui specification and make it as customisable as possible/needed by theme and by using custom components (application specific components) in it. The toolbar in my opinion should not be the holy grail for all possible toolbars, but should offer a good union way to make our admin's look similar and offer all our users a common ui pattern where e.g. the action container is always on the right side. 

@rabengraph i think this kind of JSX approaches are really nice, i already did this with all other Toolbar Items `ToolbarBackButton`, `ToolbarBreadcrumbs`, `ToolbarTitleItem` but i had to struggle a bit with styling and i did't know how to do this really well, maybe i have to talk with @jamesricky about this one more time. 

On the other side i think not everything must be in jsx, it is also a valid solution to keep this kind of things as props, especially when we pass options to this render functions (`actionItems(stackSwitchApi: StackSwitchApi)`) or someone explictly want to have them as prop (e.g. it's not the main content). In the React ecosystem there are plenty of samples where this kind of things are passed as props e.g. [Mui Button with Icons](https://material-ui.com/components/buttons/#buttons-with-icons-and-label)

Nevertheless i will give the JSX solution a second try.

Reply:
> In the React ecosystem there are plenty of samples where this kind of things are passed as prop

yes, but those don't use a function, they pass the Element directly.

as I've written here https://github.com/vivid-planet/comet-admin/pull/369#discussion_r626424082 I would remove StackSwitchApi completely

Reply:
You just supported me to removing this option, i also tended to remove `stackswitchapi` from the actionitems because they can be easily accessed by the hook. 

Mui Button and passing options should only be an example. If we abstract more things i think it is not avoidable to do make this kind of renderfunctions in props, especially when i think or look at @comet/admin  Table and customizing them. 

Don't get me wrong i also prefer JSX approach but i want to mention that the props approach is a valid one too.

Reply:
i will close this conversation for now, because main point (margin of actioncontainer) was removed, and i try out the jsx topic once again